nucypher/cli/deploy.py

```diff
@@ -93,131 +89,116 @@ def __init__(self):
         __deployment_agents = dict()
 
-        available_deployers = ", ".join(deployers)
-        click.echo("\n-----------------------------------------------")
-        click.echo("Available Deployers: {}".format(available_deployers))
-        click.echo("Blockchain Provider URI ... {}".format(config.blockchain.interface.provider_uri))
-        click.echo("Registry Output Filepath .. {}".format(config.blockchain.interface.registry.filepath))
-        click.echo("Deployer's Address ........ {}".format(config.blockchain.interface.deployer_address))
-        click.echo("-----------------------------------------------\n")
-
-        def __deploy_contract(deployer_class: ClassVar,
-                              upgradeable: bool,
-                              agent_name: str,
-                              dependant: str = None
-                              ) -> Tuple[dict, EthereumContractAgent]:
-
-            __contract_name = deployer_class._contract_name
-
-            __deployer_init_args = dict(blockchain=config.blockchain,
-                                        deployer_address=config.blockchain.interface.deployer_address)
-
-            if dependant is not None:
-                __deployer_init_args.update({dependant: __deployment_agents[dependant]})
-
-            if upgradeable:
-                secret = click.prompt("Enter deployment secret for {}".format(__contract_name),
-                                      hide_input=True, confirmation_prompt=True)
-                secret_hash = hashlib.sha256(secret)
-                __deployer_init_args.update({'secret_hash': secret_hash})
-
-            __deployer = deployer_class(**__deployer_init_args)
-
-            #
-            # Arm
-            #
-            if not force:
-                click.confirm("Arm {}?".format(deployer_class.__name__), abort=True)
-
-            is_armed, disqualifications = __deployer.arm(abort=False)
-            if not is_armed:
-                disqualifications = ', '.join(disqualifications)
-                click.secho("Failed to arm {}. Disqualifications: {}".format(__contract_name, disqualifications),
-                            fg='red', bold=True)
-                raise click.Abort()
-
-            #
-            # Deploy
-            #
-            if not force:
-                click.confirm("Deploy {}?".format(__contract_name), abort=True)
-            __transactions = __deployer.deploy()
-            __deployment_transactions[__contract_name] = __transactions
-
-            __agent = __deployer.make_agent()
-            __deployment_agents[agent_name] = __agent
+        if force:
+            deployer.blockchain.interface.registry._destroy()
 
-            click.secho("Deployed {} - Contract Address: {}".format(contract_name, __agent.contract_address),
-                        fg='green', bold=True)
+        try:
+            txhashes, agents = deployer.deploy_network_contracts(miner_secret=bytes(secrets.miner_secret, encoding='utf=8'),
```

Comment:
Lucky you, `bytes(..., encoding='utf=8')` works, but just in case:
```suggestion
            txhashes, agents = deployer.deploy_network_contracts(miner_secret=bytes(secrets.miner_secret, encoding='utf-8'),
```